tuning/data/setup_dataprocessor.py

```diff
@@ -239,9 +241,13 @@ def _process_raw_data_args(
         handlers, dataset_text_field = _get_dataset_formatting_handlers(
             data_args, packing
         )
+    elif data_args.instruction_template and data_args.response_template:
+        # Data Format 3: Chat dataset with instruction and response template
+        # We don't do processing for chat dataset
+        handlers, dataset_text_field = [], None
```

Comment:
In [this unit test](https://github.com/foundation-model-stack/fms-hf-tuning/blob/ae9182851a3a4e39d983bb41e03cbec44d1da45e/tests/test_sft_trainer.py#L868), `data_args.dataset_text_field` is not `None` so it will always satisfy elif of Line 239 instead of this elif. Hence add `data_args.dataset_text_field` to `None` in [this unit test](https://github.com/foundation-model-stack/fms-hf-tuning/blob/ae9182851a3a4e39d983bb41e03cbec44d1da45e/tests/test_sft_trainer.py#L868)

Reply:
Also have a question that if we are giving `data_args.dataset_text_field` as `None` to SFTTrainer , then how would SFTTrainer get to know that `messages` key has to be picked for data as dataset has multiple keys :

`'messages': [], 'group': 'lab_extension', 'dataset': 'base/full-extension', 'metadata': '{"num_turns": 2}'} `

Reply:
This is fixed..I noticed that SFTTrainer was not properly applying chat templates so having an explicit handler which applies chat template is an easier option...we can keep the `dataset_text_field` intact and also have more control if we need to do any handling before or after applying chat template.